### ISSUE TITLE
Install fails on iOS

### ISSUE THREAD

**Issue by daspri** [gorakhargosh/watchdog#687]
iOS does not have FSEvents afaik (it fails to compile), but watchdog forces compilation of watchdog_fsevents.c on all sys.platform ==  'darwin' systems, causing it to fail to compile/install.

Commenting that out from setup.py allows watchdog to be installed and used on iOS.

**Comment by BoboTiG** [comment]
Well, it will be possible to filter out iOS when Python will detect it. For no, I am not aware of a simple code to check if the current OS is iOS.

But I am open to suggestions, of course :)

**Comment by daspri** [comment]
platform.machine() returns the iPhone#,#/iPad#,#/Watch#,# mobile device type. ( https://gist.github.com/adamawolf/3048717 )  I believe it returns 'x86_64' on macOS (I only have a VM).

import platform
print(platform.machine() )
iPad7,5

I don't know if that is the "correct" method.  Apple is moving macOS to ARM so distinguishing ARM from i386/x86_64 in os.uname()/platform.uname() might not be reliable in the future.

print (platform.uname())

iPad:
uname_result(system='Darwin', node='xxx', release='19.5.0', version='Darwin Kernel Version 19.5.0: Tue Apr 28 22:25:26 PDT 2020; root:xnu-6153.122.1~1/RELEASE_ARM64_T8010', machine='iPad7,5', processor='arm64')

macOS VM:
uname_result(system='Darwin', node='xxx', release='19.5.0', version='Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64', machine='x86_64', processor='i386')

**Comment by BoboTiG** [comment]
Thakns, this is exactly what I was looking for :)

If you patch https://github.com/gorakhargosh/watchdog/blob/60c57aee2d39cfa8c5280cc92e3b3d278633b793/setup.py#L40, does it work for you:
```python
if sys.platform == 'darwin' and 'ipad' not in platform.machine().lower():
```
(Think to add the import too.)

**Comment by daspri** [comment]
Yes, that installs on iOS.

**Comment by BoboTiG** [comment]
Cool, do you mind opening a PR ? If so, think to add a line in the changelog § your username :)

**Comment by daspri** [comment]
Maybe if I learn how to make a PR, but I don't have time at the moment. It should be easy for anyone to fix.

**Comment by BoboTiG** [comment]
OK I will do it tomorrow then.

**Comment by BoboTiG** [comment]
@daspri could you confirm that the fix from #698 is working for you please?

**Comment by daspri** [comment]
@BoboTiG Yes, it installs on iOS (ipad).
